### PR TITLE
Fix when input directories are not found

### PR DESCRIPTION
**Pull request by odow**
x-ref: https://github.com/julia-actions/julia-processcoverage/pull/19#issuecomment-1362219782

We're probably about to have a swarm of bug reports, https://github.com/jump-dev/JuMP.jl/issues/3160, so a fix is needed ASAP.

### CODE REVIEW

**Comment by odow** [comment]
Not sure how we test this

**Comment by odow** [comment]
The work-around is https://github.com/jump-dev/JuMP.jl/pull/3161, but that doesn't scale to every package uploading coverage without an `ext` directory.

File: main.jl [hunk 1]
```diff
@@ -9,8 +9,12 @@ using CoverageTools
...
 for dir in dirs
-    isdir(dir) || error("directory \"$dir\" not found!")
+    if dir == "ext"
+        continue  # Silently skip this directory
+    elseif !isdir(dir)
+        error("directory \"$dir\" not found!")
+    end
 end
-
+dirs = filter!(isdir, dirs)
```
Comment (IanButterworth) [inline main.jl:18]:
```suggestion
filter!(isdir, dirs)
```

**Review by IanButterworth — COMMENTED** [review]
Just a nit. Seems reasonable to me

**Comment by mofii** [comment]
Yes, this PR (or similar) would be great to have merged. This is what I'm suddenly seeing on one of the packages I work on:
```
Run julia-actions/julia-processcoverage@v1
  with:
    directories: src,ext
Run julia --color=yes "$GITHUB_ACTION_PATH"/main.jl
  Activating new environment at `~/.julia/environments/coveragetempenv/Project.toml`
    Updating registry at `~/.julia/registries/General`
    Updating git-repo `https://github.com/JuliaRegistries/General.git`
   Resolving package versions...
   Installed CoverageTools ─ v1.3.0
    Updating `~/.julia/environments/coveragetempenv/Project.toml`
  [c36e975a] + CoverageTools v1.3.0
    Updating `~/.julia/environments/coveragetempenv/Manifest.toml`
  [c36e975a] + CoverageTools v1.3.0
Precompiling project...
  ✓ CoverageTools
  1 dependency successfully precompiled in [2](https://github.com/Orchard-Ultrasound-Innovation/TcpInstruments.jl/actions/runs/3753856204/jobs/6377520461#step:7:2) seconds
ERROR: LoadError: directory "ext" not found!
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:[3](https://github.com/Orchard-Ultrasound-Innovation/TcpInstruments.jl/actions/runs/3753856204/jobs/6377520461#step:7:3)3
 [2] top-level scope
   @ ~/work/_actions/julia-actions/julia-processcoverage/v1/main.jl:12
in expression starting at /home/runner/work/_actions/julia-actions/julia-processcoverage/v1/main.jl:[11](https://github.com/Orchard-Ultrasound-Innovation/TcpInstruments.jl/actions/runs/3753856204/jobs/6377520461#step:7:13)
Error: Process completed with exit code 1.
```
https://github.com/Orchard-Ultrasound-Innovation/TcpInstruments.jl/actions/runs/3753856204/jobs/6377520461

And thanks guys for making it easier for the rest of us to run the coverage tools!

**Comment by odow** [comment]
The other option is to revert the `ext` change, and force packages to opt-in to scanning `ext`.

**Review by DilumAluthge — APPROVED** [review]

**Comment by IanButterworth** [comment]
I just tagged v1.2.2 with this. Can someone confirm the fix in the wild (evidently we need better tests here!)

**Comment by DilumAluthge** [comment]
I think you'll also need to update the following tags so that they point to the same commit as v1.2.2:
- v1.2
- v1
- latest

**Comment by IanButterworth** [comment]
Ok. Perhaps we do that once the fix is confirmed?

**Comment by mofii** [comment]
I've just triggered to re-run the CI on my package, but I guess the v1 tag will need to be updated before I can pick up the change on my end?

**Comment by mofii** [comment]
I changed the settings to:
```yaml
 - uses: julia-actions/julia-processcoverage@v1.2.2
 ```
Hoping that will work 🤞

**Comment by mofii** [comment]
Yup, it's working again with tag v1.2.2!

**Comment by DilumAluthge** [comment]
@SaschaMann Should we switch `v1.2`, `v1`, and `latest` to be branches instead of tags?

**Comment by odow** [comment]
Can we just update the tags first?

This is still breaking for anyone using something like `julia-actions/julia-processcoverage@v1`.

**Comment by IanButterworth** [comment]
Sorry, was AFK for a bit. All updated now
Screenshot 2022-12-21 at 8 54 43 PM

**Comment by odow** [comment]
Thanks for the help fixing this quickly.

**Comment by SaschaMann** [comment]
We need a better CI setup for the action 😬 sorry for the inconvenience. 

> @SaschaMann Should we switch `v1.2`, `v1`, and `latest` to be branches instead of tags?

Then we'd have to push commits to several branches, wouldn't we? Updating the tags seems quicker. Might make sense to have a short script that that automatically updates the right tags when you tell it the kind of release

**Comment by fredrikekre** [comment]
Sorry for the breakage -- I just skimmed through the code and noticed the `isdir` check and simply assumed it would be ignored if not existing. Is there a reason for a hard error here regardless? Seems like silently ignoring missing directories might be more useful, e.g. if you want to have the same (centrally managed or w/e) configuration for multiple repositories.

**Comment by SaschaMann** [comment]
The foreshadowing is real: https://github.com/julia-actions/julia-processcoverage/pull/8#issuecomment-854993035

Some reasons for why this should be an error here: https://github.com/julia-actions/julia-processcoverage/pull/8#issuecomment-855017704

---

> Seems like silently ignoring missing directories might be more useful, e.g. if you want to have the same (centrally managed or w/e) configuration for multiple repositories.

We could add an input that changes it to a warning. Or change the default and make errors opt-in. Not sure what's better.

**Pull request merged**